hydromt_wflow/wflow.py

```diff
@@ -848,204 +849,370 @@ def setup_laimaps(self, lai_fn="modis_lai"):
+        )
+        gdf["fid"] = ids_out.astype(np.int32)
+        self.set_staticgeoms(gdf, name="gauges")
+        self.logger.info(f"Gauges map based on catchment river outlets added.")
+
+        self.setup_config_output_timeseries(
+            mapname="wflow_gauges",
+            toml_output=toml_output,
+            header=gauge_toml_header,
+            param=gauge_toml_param,
+        )
+
     def setup_gauges(
         self,
-        gauges_fn="grdc",
-        source_gdf=None,
-        index_col=None,
-        snap_to_river=True,
-        mask=None,
-        derive_subcatch=False,
-        derive_outlet=True,
-        basename=None,
-        update_toml=True,
-        gauge_toml_header=None,
-        gauge_toml_param=None,
+        gauges_fn: Union[str, Path, gpd.GeoDataFrame],
+        index_col: Optional[str] = None,
+        snap_to_river: Optional[bool] = True,
+        mask: Optional[np.ndarray] = None,
+        snap_uparea: Optional[bool] = False,
+        max_dist: Optional[float] = 10e3,
+        wdw: Optional[int] = 3,
+        rel_error: Optional[float] = 0.05,
+        derive_subcatch: Optional[bool] = False,
+        basename: Optional[str] = None,
+        toml_output: Optional[str] = "csv",
+        gauge_toml_header: Optional[List[str]] = ["Q", "P"],
+        gauge_toml_param: Optional[List[str]] = [
+            "lateral.river.q_av",
+            "vertical.precipitation",
+        ],
         **kwargs,
     ):
-        """This components sets the default gauge map based on basin outlets and additional
-        gauge maps based on ``gauges_fn`` data.
+        """This components sets a gauge map based on ``gauges_fn`` data.
 
         Supported gauge datasets include "grdc"
         or "<path_to_source>" for user supplied csv or geometry files with gauge locations.
         If a csv file is provided, a "x" or "lon" and "y" or "lat" column is required
-        and the first column will be used as IDs in the map. If ``snap_to_river`` is set
-        to True, the gauge location will be snapped to the boolean river mask. If
-        ``derive_subcatch`` is set to True, an additional subcatch map is derived from
+        and the first column will be used as IDs in the map.
+
+        There are three available methods to prepare the gauge map:
+
+        * no snapping: ``mask=None``, ``snap_to_river=False``, ``snap_uparea=False``.
+          The gauge locations are used as is.
+        * snapping to mask: the gauge locations are snapped to a boolean mask map: either provide
+          ``mask`` or set ``snap_to_river=True`` to snap to the river (default). ``max_dist`` can be
+          used to set the maximum distance to snap to the mask.
+        * snapping based on upstream area matching: : ``snap_uparea=True``. The gauge locations
+          are snapped to the closest matching upstream area value. Requires gauges_fn to have
+          an `uparea` [m] column. The closest value will be looked for in a cell window of size ``wdw``
```

Comment:
I think this is the last comment: uparea unit should be [km2], not [m], right?

Reply:
Indeed you're right! I changed it now and also saw I forgot the docstring in the workflow function... should be all good now!